server/src/domain/person/person.service.ts

```diff
@@ -98,16 +98,34 @@ export class PersonService {
     return this.repository.create({ ownerId: auth.user.id });
   }
 
-  async reassignFaces(auth: AuthDto, personId: string, dto: AssetFaceUpdateDto): Promise<PersonResponseDto[]> {
-    await this.access.requirePermission(auth, Permission.PERSON_WRITE, personId);
+  async reassignFace(authUser: AuthDto, personId: string, dto: FaceDto): Promise<PersonResponseDto> {
+    await this.access.requirePermission(authUser, Permission.PERSON_WRITE, personId);
+
+    await this.access.requirePermission(authUser, Permission.PERSON_CREATE, dto.id);
+    const face = await this.repository.getFaceById(dto.id);
+    const person = await this.findOrFail(personId);
+
+    await this.repository.reassignFace(face.id, personId);
+    if (person.faceAssetId === null) {
+      await this.createNewFeaturePhoto([person.id]);
+    }
+    if (face.person && face.person.faceAssetId === face.id) {
+      await this.createNewFeaturePhoto([face.person.id]);
+    }
```

Comment:
I am confused a bit. The `createNewFeaturePhoto` argument takes in a list of faces, but you call it two separate times here. Shouldn't you instead start with a list like `refreshIds = []` and then add to the list in the if statements and then finally call `this.createNewFeaturePhoto` with the list?